src/parser/ast.rs

```diff
@@ -549,13 +549,19 @@ impl<'a> Ast<'a> {
     }
 
     pub fn if_expr(condition: Ast<'a>, then: Ast<'a>, otherwise: Option<Ast<'a>>, location: Location<'a>) -> Ast<'a> {
-        Ast::If(If {
+        
+ 
+        match &otherwise{
+            Some(_ast) => Ast::If(If {
             condition: Box::new(condition),
             then: Box::new(then),
             otherwise: otherwise.map(Box::new),
```

Comment:
If you're explicitly matching on `otherwise` here now, then you can use the `_ast` directly instead of ignoring it and doing `otherwise.map(...)` later on.